### DISCUSSION TITLE
Zeek worker crashes with (cannot allocate memory) error

### DISCUSSION THREAD

**Discussion by samanosuke26** [Security-Onion-Solutions/securityonion#8696] [Unsupported Versions]
Security Onion version 2.3.130
zeek 4.0.7
OS Centos 7
uname -a : Linux 3.10.0-1160.66.1.e17.x86_64

Despite the container having no resource problems the worker keeps crashing with
"fatal error: problem with interface af_pcaket::bond0 (cannot allocate memory)
I’ve been unable to figure out why. I watch docker stats and the memory usage never goes above 4%
Any help would be great.

**Comment by reyesj2** [comment]
Do you have custom configurations to Zeek? Try rolling to default configuration for changes. Any clues in Zeek logs before crash?
Can you check that the memory usage of not only Zeek is low, but your overall system memory usage?

Check grafana for some insight into what could be going on

**Reply by samanosuke26** [reply]
Reyesj2 Thanks for the reply,
I wasn't the engineer but I checked my salt-master at /opt/so/saltstack/local/salt/zeek/policy/custom and saw nothing. 
Zeek logs only show the allocate memory error and termination. No help, really.
As for the system memory usage, I watched free -m as the container crashed and there was still about 10Gb of the 15Gb RAM available.

**Comment by dougburks** [comment]
How much traffic are you monitoring?

What are the specs of your system?

Have you checked the hardware requirements?
https://docs.securityonion.net/en/2.3/hardware.html#sensor-hardware-considerations

Have you tried updating to Security Onion 2.3.160? It includes a Zeek update:
https://blog.securityonion.net/2022/08/security-onion-23160-now-available.html

**Reply by samanosuke26** [reply]
I'm using many different sensors on different taps but the one I'm currently using as my troubleshooting base has an average monitoring load of 2.63 Mbps
4 CPU cores
16 GB RAM
1 TB storage

Yes we go off the HW requirements when building them

No I haven't tried upgrading. This is on an out-of-band network therefore upgrades are a planned event. We just recently upgraded from 2.3.90 --> 2.3.130

**Reply by dougburks** [reply]
Just to confirm...this a dedicated sensor connected to a separate manager, right?

As mentioned in my previous reply, Security Onion 2.3.160 includes a Zeek update and this Zeek update fixes several issues in Zeek:
https://github.com/zeek/zeek/releases/tag/v4.0.8

So we'd recommend that you go ahead and plan an upgrade and see if that helps.
